### DISCUSSION TITLE
[RFC] "Multi-Streaming" in CUDA Graphs

### DISCUSSION THREAD

**Discussion by ezhulenev** [openxla/xla#4318] [Show and tell]
Hi everyone,

We are working on adding multi-streaming to CUDA graphs in XLA. The high level plan is capture in this [document](https://docs.google.com/document/d/1QIQogiH9Qyp_DxoNocBIphEMJH4ctIeiztSPKAc2Dzs/edit).

**Comment by MoFHeka** [comment]
Any progress?

**Comment by ezhulenev** [comment]
It's behind the flag today: https://github.com/openxla/xla/blob/main/xla/xla.proto#L615-L616, we are working on more explicit multi-streaming + cuda graphs support on XLA, it's here: https://github.com/openxla/xla/blob/main/xla/service/gpu/execution_stream_assignment.h

**Reply by MoFHeka** [reply]
Thanks for your reply. May I ask how can I confirm the concurrence regions have been assigned? The after optimized hlo graph (html) seems no different between the flag on and off.

**Reply by ezhulenev** [reply]
you can run with `TF_CPP_VMODULE=gpu_command_buffer=100`, it will dump all CUDA graphs to logs (and temp files), they are dot files and you can plot them to see if kernels form a DAG
